tests/testthat/test-Logger.R

```diff
@@ -1,240 +1,468 @@
+    )
+    expect_warning(
+      logger$log_warn("test console and filewriting", tic = logger$start_time),
+      glue::glue("{ts_str} - {Sys.info()[['user']]} - WARNING - test console and filewriting")
+    )
+    expect_error(
+      logger$log_error("test console and filewriting", tic = logger$start_time),
+      glue::glue("{ts_str} - {Sys.info()[['user']]} - ERROR - test console and filewriting")
+    )
+
+
+    # Test logging to file has the right formatting and message type
     expect_true(logger$log_filename %in% dir(log_path))
-    expect_equal(readLines(file.path(log_path, logger$log_filename)),
-                 glue::glue("{ts_str} - {Sys.info()[['user']]} - INFO - test - filewriting"))
+    expect_equal(
+      readLines(logger$log_realpath),
+      c(
+        glue::glue("{ts_str} - {Sys.info()[['user']]} - INFO - test console and filewriting"),
+        glue::glue("{ts_str} - {Sys.info()[['user']]} - WARNING - test console and filewriting"),
+        glue::glue("{ts_str} - {Sys.info()[['user']]} - ERROR - test console and filewriting")
+      )
+    )
 
+
+    # Test logging to db writes to the correct fields
     logger$log_to_db(n_insertions = 13)
     expect_equal(nrow(log_table_id), 2)
     expect_equal(dplyr::pull(log_table_id, "n_insertions"), c(42, 13))
+
     logger$log_to_db(n_deactivations = 37)
     expect_equal(nrow(log_table_id), 2)
     expect_equal(dplyr::pull(log_table_id, "n_deactivations"), c(60, 37))
 
-    # Cleanup
-    dir(log_path) |>
-      purrr::keep(~ endsWith(., ".log")) |>
-      purrr::walk(~ unlink(file.path(log_path, .)))
 
+    # Clean up
     connection_clean_up(conn)
+    file.remove(logger$log_realpath)
+    rm(logger)
+    invisible(gc())
   }
 })
 
-test_that("Logger stops if file exists", {
 
-  db_tablestring <- "test.SCDB_logger"
-  ts <- Sys.time()
-  log_path <- tempdir()
-  logger <- Logger$new(
-    db_tablestring = db_tablestring,
-    ts = ts,
-    log_table_id = NULL,
-    log_path = log_path
-  )
+test_that("Logger: file logging stops if file exists", {
 
-  utils::capture.output(logger$log_info("test message"))
+  # Set options for the test
+  log_path <- tempdir(check = TRUE)
+  db_table <- "test.SCDB_logger"
+  timestamp <- Sys.time()
+
+  # Create logger1 and logger2 which uses the same file
+  # Since start_time is the same for both
+  logger1 <- Logger$new(
+    db_table = db_table,
+    timestamp = timestamp,
+    log_path = log_path,
+    output_to_console = FALSE
+  )
 
   logger2 <- Logger$new(
-    db_tablestring = db_tablestring,
-    ts = ts,
-    start_time = ts,
-    log_table_id = NULL,
-    log_path = log_path
+    db_table = db_table,
+    timestamp = timestamp,
+    start_time = timestamp,
+    log_path = log_path,
+    output_to_console = FALSE
   )
 
+
+  # logger1 should be able to successfully write
+  logger1$log_info("test message")
+
+
+  # whereas logger2 should fail since the log file now exists
   expect_error(
```

Comment:
I cannot figure out where this error is created. 
I suggest that the functionality is changed so that the initializer checks if the log_realpath file exists on disk if log_path is not null. That will make it safer and easier to debug during use.

Reply:
In `Logger`s active binding: `log_filename` there is this code:

```
if (file.exists(file.path(self$log_path, private$.log_filename))) {
    stop(sprintf("Log file '%s' already exists!", private$.log_filename))
}
```

The file is not created until a `$log_info/warn/error()` function is called. 
We could create the file on initialization and then do the checks as you suggest.
But the current implementation pospones this check as late as possible, which I think is the better choice. If we create the file at initialization, we may end up with a lot of empty log files (since a log file is created every time we create a `Logger`, even if we do not use it to log anything)